providers/class.two-factor-provider.php

```diff
@@ -45,6 +45,18 @@ public function print_label() {
 	abstract function authentication_page( $user );
 
 	/**
+	 * Allow providers to process the two factor authentication request for
+	 * additional tasks such as re-sending tokens, etc.
+	 *
+	 * @param WP_User $user WP_User object of the logged-in user.
+	 *
+	 * @return boolean Return `true` to prevent further processing.
+	 */
+	public function process_request( $user ) {
```

Comment:
Should the function name here be `pre_process_request` or the like to acknowledge it runs first and can short out the other processing?